forms.rst

```diff
@@ -24,22 +24,22 @@ install the form feature before using it:
 
     $ composer require symfony/form
 
-.. note::
+Usage
+-----
 
-    The Symfony Form component is a standalone library that can be used outside
-    of Symfony projects. For more information, see the
-    :doc:`Form component documentation </components/form>` on GitHub.
+The recommended workflow when working with Symfony forms is the following:
 
-.. index::
-   single: Forms; Create a simple form
+* **Build the form** in a Symfony controller or using a dedicated form class;
+* **Render the form** in a template so the user can edit and submit it;
+* **Process the form** to store the submitted data (e.g. in a database);
+* **Validate the form** to check that the submitted data is correct.
```

Comment:
While the order is correct, the sentences are a bit weird as it now seems we save the data to the database and then validate it. I would do something like:

```rst
#. **Build the form** in a Symfony controller or using a dedicated form class;
#. **Render the form** in a template so the user can edit and submit it;
#. **Process the form** to PHP values;
#. **Validate the form** to check that the submitted data is correct;
```

(maybe with another extra step e.g. "Act on the submitted data (e.g. store it in a database).")

Reply:
You are right ... but I think the real problem was separating "process" and "validate" ... which are closely related. In the docs we're keeping them in separate sections, but in this list I propose to change it like this:

```rst
#. **Build the form** in a Symfony controller or using a dedicated form class;
#. **Render the form** in a template so the user can edit and submit it;
#. **Process the form** to validate the submitted data, transform it into PHP
   data and do something with it (e.g. persist it in a database).
```